subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c

```diff
@@ -8,12 +8,161 @@
 #include <zephyr/kernel.h>
 #include <zephyr/logging/log.h>
 #include <modem/modem_jwt.h>
+#if defined(CONFIG_NRF_MODEM_LIB)
 #include <nrf_modem_at.h>
+#endif
+#include <string.h>
+#include <net/nrf_jwt.h>
+#include <zephyr/sys/base64.h>
+#include <psa/crypto.h>
+
+/* For testing */
+#define READ_FROM_INC_FILE 0
+#if READ_FROM_INC_FILE
+#include CONFIG_NRF_CLOUD_CERTIFICATES_FILE
+#endif
 
 #define GET_TIME_CMD "AT%%CCLK?"
 
 LOG_MODULE_REGISTER(nrf_cloud_jwt, CONFIG_NRF_CLOUD_LOG_LEVEL);
 
+#define PRV_KEY_SZ (NRF_JWT_KEY_DER_SZ)
+#define PRV_KEY_DER_SZ (138)
+#define PRV_KEY_PEM_SZ (256)
+#define PRV_KEY_DER_START_IDX (36)
+
+#if defined(CONFIG_NRF_CLOUD_JWT_GEN_METHOD_PSA)
+#define JWT_KEY_ID CONFIG_NRF_CLOUD_JWT_PSA_KEY_ID
+#else
+#define JWT_KEY_ID PSA_KEY_ID_NULL
+#endif
+
+static void remove_newlines(char *const str)
+{
+	size_t new;
+	size_t len = strlen(str);
+
+	for (size_t old = new = 0; old < len; ++old) {
+		if (str[old] != '\n') {
+			str[new++] = str[old];
+		}
+	}
+
+	str[new] = '\0';
+}
+
+#define BEGIN_PRV_KEY "-----BEGIN PRIVATE KEY-----"
+#define END_PRV_KEY "-----END PRIVATE KEY-----"
+static int strip_non_key_data(char *const str)
+{
+	char *start;
+	char *end;
+	size_t new_len;
+
+	/* Find and remove end string */
+	end = strstr(str, END_PRV_KEY);
+	if (end == NULL) {
+		return -EINVAL;
```

Comment:
This will fail for non-PEM credentials, which we may or may not want to have at our disposal. It saves RAM, at least